### PR TITLE
spells activity UI added

### PR DESCRIPTION
**Pull request by benjaglobant**
In this PR I have implemented the SpellsActivity UI using a recycler view to show the list of spells (mocked in this PR, in the next PR I will implement the Spells request to the API.
HarryPotterApp gif (normal device):
spellsUIPRJ7
(smaller device):
spellsUIPRNexusS

### CODE REVIEW

File: app/src/main/java/com/globant/harrypotterapp/activity/SpellsActivity.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    private fun updateUI(spells: ListOfSpellsMocked) {
+        spellsAdapter.submitList(spells.listOfSpells)
+        binding.spellsActivityRecyclerView.layoutManager = LinearLayoutManager(this)
+        binding.spellsActivityRecyclerView.adapter = spellsAdapter
+    }
+
+    companion object {
+        fun getIntent(activity: Activity): Intent = Intent(activity, SpellsActivity::class.java)
```
Comment (mauroLonghi) [inline app/src/main/java/com/globant/harrypotterapp/activity/SpellsActivity.kt]: Please change the type of the paramter to a Context

File: app/src/main/java/com/globant/harrypotterapp/contract/SpellsContract.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.globant.harrypotterapp.contract
```
Comment (SartiMau) [inline app/src/main/java/com/globant/harrypotterapp/contract/SpellsContract.kt]: move this package inside the viewmodel package pls

Comment (benjaglobant) [reply app/src/main/java/com/globant/harrypotterapp/contract/SpellsContract.kt]: Done!

**Review by BQuinteros — APPROVED** [review]

File: app/src/main/java/com/globant/harrypotterapp/activity/SpellsActivity.kt (hunk 1 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/harrypotterapp/activity/SpellsActivity.kt]: Done!

File: app/src/main/java/com/globant/harrypotterapp/activity/SpellsActivity.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        binding = ActivitySpellsBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        spellsViewModel.getSpellsLiveData().observe(::getLifecycle, ::updateUI)
+        spellsViewModel.fetchSpells()
+    }
+
+    private fun updateUI(spells: SpellsData) {
+        when (spells.state) {
+            SpellsState.SHOW_LOADER -> binding.spellsActivityLoader.visibility = View.VISIBLE
+            SpellsState.SHOW_DATA -> {
```
Comment (SartiMau) [inline app/src/main/java/com/globant/harrypotterapp/activity/SpellsActivity.kt]: to make the code more readable you can put this lines in a method called showSpellsData(..) or showData(..)

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/globant/harrypotterapp/activity/SpellsActivity.kt (hunk 3 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/harrypotterapp/activity/SpellsActivity.kt]: Yes you are right! Thanks Mau, done!

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**